### ISSUE TITLE
linter: jsx-a11y/role-supports-aria-props incorrect for role=combobox and aria-haspopup attribute

### ISSUE THREAD

**Issue by jackstevenson** [oxc-project/oxc#21719]
### What version of Oxlint are you using?

1.61.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?
```json
{
  "plugins": ["jsx-a11y"],
  "categories": {
    "correctness": "off"
  },
  "rules": {
    "jsx-a11y/role-supports-aria-props": "error"
  }
}
```

### What happened?

Oxlint incorrectly reports the error:
```
The attribute `aria-haspopup` is not supported by the role `combobox`.
```

[From MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/combobox_role#description)
> Comboboxes have an implicit [aria-haspopup](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-haspopup) value of listbox, so including this attribute is optional if the popup is a listbox. If the combobox popup element is a tree, grid, or dialog (anything other than a listbox), the aria-haspopup attribute is required. The value of aria-haspopup must be either the tree, grid, dialog, or listbox role. Note that for this property, true means menu, so make sure that the value corresponds to the role of the popup, not a Boolean value.

[Playground repro](https://playground.oxc.rs/?lintRules=jsx-a11y%2Frole-supports-aria-props&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22jsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%22config%22%3A%22%7B%5C%22categories%5C%22%3A%7B%5C%22correctness%5C%22%3A%5C%22off%5C%22%7D%2C%5C%22rules%5C%22%3A%7B%5C%22jsx-a11y%2Frole-supports-aria-props%5C%22%3A%5C%22error%5C%22%7D%2C%5C%22plugins%5C%22%3A%5B%5C%22eslint%5C%22%2C%5C%22oxc%5C%22%2C%5C%22typescript%5C%22%2C%5C%22unicorn%5C%22%2C%5C%22jsx-a11y%5C%22%5D%7D%22%7D%2C%22formatter%22%3A%7B%22useTabs%22%3Afalse%2C%22tabWidth%22%3A2%2C%22endOfLine%22%3A%22lf%22%2C%22printWidth%22%3A80%2C%22singleQuote%22%3Afalse%2C%22jsxSingleQuote%22%3Afalse%2C%22quoteProps%22%3A%22as-needed%22%2C%22trailingComma%22%3A%22all%22%2C%22semi%22%3Atrue%2C%22arrowParens%22%3A%22always%22%2C%22bracketSpacing%22%3Atrue%2C%22bracketSameLine%22%3Afalse%2C%22objectWrap%22%3A%22preserve%22%2C%22singleAttributePerLine%22%3Afalse%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%7D%2C%22isolatedDeclarations%22%3A%7B%22stripInternal%22%3Afalse%7D%2C%22codegen%22%3A%7B%22normal%22%3Atrue%2C%22jsdoc%22%3Atrue%2C%22annotation%22%3Atrue%2C%22legal%22%3Atrue%7D%2C%22compress%22%3A%7B%7D%2C%22mangle%22%3A%7B%22topLevel%22%3Atrue%2C%22keepNames%22%3Afalse%7D%2C%22controlFlow%22%3A%7B%22verbose%22%3Afalse%7D%2C%22inject%22%3A%7B%22inject%22%3A%7B%7D%7D%2C%22define%22%3A%7B%22define%22%3A%7B%7D%7D%7D&code=const+DummyComponent+%3D+%28%29+%3D%3E+%7B%0A++return+%28%0A++++%3Cinput+type%3D%22text%22+role%3D%22combobox%22+aria-haspopup%3D%22listbox%22+%2F%3E%0A++%29%0A%7D%0A%0Aexport+default+DummyComponent)

**Comment by jackstevenson** [comment]
It appears `AriaProperty::HasPopup` is missing from `COMBOBOX_PROPS` in https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/jsx_a11y/role_supports_aria_props.rs#L401-L402
